### DISCUSSION TITLE
Sort `packageRules` options in docs

### DISCUSSION THREAD

**Discussion by lindhe** [renovatebot/renovate#29562] [Suggest an Idea]
[In the docs](https://docs.renovatebot.com/configuration-options/#packagerules), the options under `packageRules` appears to not be sorted in alphabetical order:

![image](https://github.com/renovatebot/renovate/assets/7773090/6e5a97fc-862a-4df5-8469-bc6179cb40c2)

I suggest we change the order so they are sorted, to make docs easier to navigate. Would a PR be welcome?

**Comment by viceice** [comment]
sounds good, we should do it for all sub sections there and validate via docs test

**Reply by rarkins** [reply]
I would also like to split the docs into a separate page, not sure if it should be done first, at the same time, or after

**Reply by HonkingGoose** [reply]
I agree that we should have tests to confirm the docs have a logical order! 😉 

## Split docs first, at the same time, or after?

> I would also like to split the docs into a separate page, not sure if it should be done first, at the same time, or after

Hmm. Spltting the `packageRules` to a separate page is a big change.

How about refactoring and testing _first_, and then making the big move in a _second_ PR? Here's how I think it could work:

### Write tests and refactor current `packageRules` docs PR

1. Write test to check if current `packageRules` entries are in the right order, test fails, because the list is not properly sorted right now.
2. Move `packageRules` items into correct order, test passes now.
3. Merge "Test and move items PRs".

### Move `packageRules` to new page PR

4. Split `packageRules` content off to a new page (`docs/usage/package-rules.md` or whatever name is best).
5. Add short and simple introductory text about the page, explaining what it is about, what you can do with it and so on.
6. Point test to new file. Confirm `packageRules` items are in the correct order.
7. Update all the links that point to `packageRules` items (will be a _lot_!). I think we already have docs test that will tell us about the now broken relative links. So I think we can use the tests to guide us to the correct links.
8. Merge PR.

## Let me know what you think!

As usual, let me know what you think of my idea. 😉

**Reply by HonkingGoose** [reply]
Both `@viceice` and `@rarkins` gave my idea a thumbs up, so here are the issues to track this work:

- #29703
- #29704
